E/elmerfem/build_tarballs.jl

```diff
@@ -0,0 +1,151 @@
+#
+#   The contents of cmake_install.cmake lines 40-42 are:
+#
+#       if("x${CMAKE_INSTALL_COMPONENT}x" STREQUAL "xUnspecifiedx" OR NOT CMAKE_INSTALL_COMPONENT)
+#           file(INSTALL DESTINATION "${CMAKE_INSTALL_PREFIX}/bin" TYPE FILE FILES "/workspace/srcdir/elmerfem/-L/workspace/destdir/bin -lblastrampoline-5")
+#       endif()
+#   
+#   The linux cmake_install.cmake doesn't include this line at all, so perhaps we can just remove it
+#   before running `make`...?
+# XXX: build fails on FreeBSD with compilation error... suggests we need to use an older C++ standard:
+#       [17:35:11] /workspace/srcdir/elmerfem/meshgen2d/src/BGTriangleMesh.cpp:33:7: error: no member named 'random_shuffle' in namespace 'std'
+#       [17:35:11]         std::random_shuffle(indirect, indirect + len);
+#       [17:35:11]         ~~~~~^
+#       [17:35:11] 1 error generated.
+platforms = [
+    p for p in expand_gfortran_versions(supported_platforms()) if (
+        libgfortran_version(p) >= v"5" &&
+        !Sys.isapple(p) &&
+        !(arch(p) in ("armv6l", "armv7l")) &&
+        !Sys.iswindows(p) &&
+        !Sys.isfreebsd(p)
+    )
+]
+
+# The products to which we provide easy access in the jll wrapper.
+products = [
+    ExecutableProduct("ElmerGrid", :elmer_grid),
+    ExecutableProduct("ElmerSolver", :elmer_solver),
+    ExecutableProduct("Mesh2D", :mesh2d),
+    ExecutableProduct("Radiators", :radiators),
+    ExecutableProduct("ViewFactors", :view_factors),
+    ExecutableProduct("elmerf90", :elmerf90),
+    ExecutableProduct("elmerld", :elmerld),
+    ExecutableProduct("matc", :matc),
+]
+
+# Dependencies that must be installed before this package can be built.
+# We require CompilerSupportLibraries for the user to have e.g. libgfortran after
+# installing this package.
+# In addition, we use LLVM OpenMP on BSD systems (OpenBSD & MacOS).
+dependencies = [
+    Dependency(
+        PackageSpec(;
+            name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"
+        ),
+    ),
+    Dependency(
+        PackageSpec(; name="LLVMOpenMP_jll", uuid="1d63c593-3942-5779-bab2-d838dc0a180e");
+        platforms=filter(Sys.isbsd, platforms),
+    ),
+    Dependency(
+        PackageSpec(;
+            name="libblastrampoline_jll", uuid="8e850b90-86db-534c-a0d3-1478176c7d93"
+        );
+        compat="5.4",
+    ),
+]
+
+# ElmerSolver and other binaries need to find the items under X/share/elmerfem,
+# where X is the install root.
+# The installation root is embedded into the binary during compilation, however this isn't
+# the correct path at runtime. Hence we instead ensure that ELMER_HOME is set when we try
+# to run the binaries.
+init_block = raw"""ENV["ELMER_HOME"] = artifact_dir"""
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(
+    ARGS,
+    name,
+    version,
+    sources,
+    script,
+    platforms,
+    products,
+    dependencies;
+    julia_compat="1.6",
```

Comment:
libblastrampoline 5.4 requires julia v1.9: https://github.com/JuliaLang/julia/blob/v1.9.0/stdlib/libblastrampoline_jll/Project.toml (1.8 had 5.1.1: https://github.com/JuliaLang/julia/blob/v1.8.0/stdlib/libblastrampoline_jll/Project.toml).  Are you fine with requiring julia v1.9?
```suggestion
    julia_compat="1.9",
```
Otherwise supporting different versions of libblastrampoline before 5.4 is a pain.

Reply:
Yep Julia >=1.9 is fine for us, thanks!